components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/auth/OIDCOauthManager.java

```diff
@@ -175,30 +174,100 @@ private JSONObject retrieveClientCredentials(String dcrEp, HttpClient client) th
         }
     }
 
+    /**
+     * This method will be used to validate access token.
+     *
+     * @param token This will be the access token
+     * @return the boolean status of the validity of the access token
+     */
     public boolean validateToken(String token) throws OIDCProviderException {
         try {
-            Unirest.setHttpClient(Utils.getAllSSLClient());
-            com.mashape.unirest.http.HttpResponse<String> stringResponse
-                    = Unirest.post(CelleryConfig.getInstance().getIdpURL() + Constants.INTROSPECT_ENDPOINT)
-                    .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_FORM_URLENCODED)
-                    .basicAuth(CelleryConfig.getInstance().getIdpAdminUsername()
-                            , CelleryConfig.getInstance().getIdpAdminPassword()).body("token=" + token).asString();
-
-            JSONObject jsonResponse = new JSONObject(stringResponse.getBody());
-            if (stringResponse.getStatus() != 200) {
+            HttpResponse response = makeRequestforTokenValidition(token);
+            StringBuilder builder = getResponseString(response);
+            JSONObject jsonObject = new JSONObject(builder.toString());
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (statusCode >= 200 && statusCode < 400) {
```

Comment:
Shouldn't this check change ? This check is for successful response. IINM it should be the error response according to the if block body